app/models/miq_server/role_management.rb

```diff
@@ -46,6 +46,7 @@ def set_active_role_flags
     self.has_active_userinterface = self.has_active_role?("user_interface")
     self.has_active_websocket     = self.has_active_role?("websocket")
     self.has_active_webservices   = self.has_active_role?("web_services")
+    self.has_active_cockpit_ws    = MiqCockpitWsWorker.can_activate_cockpit?
```

Comment:
The goal here was to make sure that if the server assigned the relevant role doesn't have cockpit-ws and cockpit-ssh installed the UI would fall back to the current functionality. So I wanted to make sure that only servers that actually have the correct software installed are marked as active. That means checking that cockpit-ws and cockpit-ssh are in fact installed.